File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
let map;

let inOrder = (root,level) => {
    if (root == null)
        return [];
    if(!(level in map))
        map[level] = [];
    map[level].push(root.val);
    return [...inOrder(root.left, level+1), root.val, ...inOrder(root.right, level+1)];
}

var deepestLeavesSum = function(root) {
    map = {};
    inOrder(root,0);
    return map[Math.max(...Object.keys(map))].reduce((prev,curr) => prev+curr,0)
};

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int deepest = 0, sum = 0;
int deepestLeavesSum(TreeNode* root, int depth = 0) {
	if(!root) return 0;  
	if(!root -> left && !root -> right) 
		if(depth == deepest) sum += root -> val;  
		else if(depth > deepest) sum = root -> val, deepest = depth; 
	deepestLeavesSum(root -> left, depth + 1);   
	deepestLeavesSum(root -> right, depth + 1);  
	return sum; 
}
};